jwcrypto/jwk.py

```diff
@@ -367,22 +424,19 @@ def _import_pyca_pub_ec(self, key, **params):
         self.import_key(**params)
 
     def _generate_OKP(self, params):
-        crv = 'Ed25519'  # Ed25519 is the default Edwards curve
-        if 'curve' in params:
-            crv = params.pop('curve')
-        # 'curve' is for backwards compat, if 'crv' is defined it takes
-        # precedence
-        if 'crv' in params:
-            crv = params['crv']
-        # Make crv available in params for _import_pyca_*_okp() methods
-        params['crv'] = crv
-        if crv == 'Ed25519':
-            key = ed25519.Ed25519PrivateKey.generate()
-        elif crv == 'Ed448':
-            key = ed448.Ed448PrivateKey.generate()
+        if 'crv' not in params:
+            raise InvalidJWKValue('Must specify "crv" for OKP key generation')
+        if params['crv'] == 'Ed25519':
```

Comment:
This ``if/elif/elif`` approach is hard to maintain. How about an internal mapping from name to type?

```
_CURVE_CLASSES = {
   'Ed25519': (Ed25519PublicKey, Ed25519PrivateKey),
   ...
}
```

Then you can replace ``name in ['Ed25519', 'Ed448', 'X25519', 'X448']`` with ``name in _CURVE_CLASSES`` and the lookup with ``pubclass, privclass = _CURVE_CLASSES.get(params['crv'], (None, None))`` and a check like ``if pubclass is not None: return pubclass.generate()``.

Reply:
Yeah didn't like the if/then/else myself, I will try to implement your suggestion and see if it looks better as it sounds promising.

Reply:
Please check latest rebase, I added a patch to use a lookup table everywhere possible